packages/components/Button/src/FAB/FABCore.tsx

```diff
@@ -65,20 +66,54 @@ export const FAB = compose<FABType>({
       }
       const label = accessibilityLabel ?? childText;
 
-      const fabWithoutShadow = (
-        <Slots.root {...mergedProps} accessibilityLabel={label}>
+      const buttonContent = (
+        <React.Fragment>
           {icon && <Slots.icon {...iconProps} />}
-          {React.Children.map(children, (child) =>
-            typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
-          )}
-        </Slots.root>
+          {showChildren &&
+            React.Children.map(children, (child) =>
+              typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
+            )}
+        </React.Fragment>
       );
 
       const hasShadow = Platform.OS === 'ios';
+      const hasRipple = Platform.OS === 'android';
+
       if (hasShadow) {
-        return <Slots.shadow>{fabWithoutShadow}</Slots.shadow>;
+        return (
+          <Slots.shadow>
+            <Slots.root {...mergedProps} accessibilityLabel={label}>
+              {buttonContent}
+            </Slots.root>
+          </Slots.shadow>
+        );
+      } else if (hasRipple) {
+        const [extractedMargin, styleWithoutMargin] = extractMarginAndroid(mergedProps.style);
+        return (
+          <Slots.root style={extractedMargin}>
+            <Slots.ripple
```

Comment:
There is some duplicated logic between extracting the margin in the shadow slot and here in the FAB. I spoke offline about this briefly. My proposal is this: Could we do something where we extract the "RootWithRipple" into it's own slot. I think as is right now, this logic gets very complicated, especially if we need to duplicate it for any components that need either a shadow or ripple